cnd/src/btsieve/bitcoin/bitcoind_connector.rs

```diff
@@ -76,20 +77,23 @@ impl BlockByHash for BitcoindConnector {
     }
 }
 
-pub async fn chain_info(conn: &BitcoindConnector) -> Result<ChainInfo, anyhow::Error> {
-    let client = conn.client.clone();
-    let chaininfo_url = conn.chaininfo_url.clone();
+#[async_trait]
+impl FetchNetworkId<Network> for BitcoindConnector {
+    async fn network_id(&self) -> anyhow::Result<Network> {
+        let client = self.client.clone();
+        let chaininfo_url = self.chaininfo_url.clone();
 
-    let chain_info = client
-        .get(chaininfo_url)
-        .send()
-        .await?
-        .json::<ChainInfo>()
-        .await?;
+        let chain_info: ChainInfo = client
```

Comment:
Is it really needed, I believe that `.json::<ChainInfo>()` already hints the compiler.

Reply:
The compiler figures it out fine but the IDE plugin struggles. Thought I would leave it in for developer ergonomics.